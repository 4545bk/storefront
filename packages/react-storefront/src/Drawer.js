/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component } from 'react'
import ResizeObserver from 'resize-observer-polyfill'
import MUIDrawer from '@material-ui/core/Drawer'
import { withStyles } from '@material-ui/core/styles';
import Close from '@material-ui/icons/Close'
import Button from '@material-ui/core/Button'
import PropTypes from 'prop-types'
import Typography from '@material-ui/core/Typography'
import { inject } from 'mobx-react'
import classnames from 'classnames'

/**
 * A slide-in drawer with fab close button.
 */
export const styles = theme => ({
  closeButton: {
    position: 'absolute',
    right: '10px',
    top: '-28px',
    zIndex: 1
  },

  paper: {
    overflowY: 'visible'
  },

  title: {
    textAlign: 'center',
    padding: '20px',
    fontSize: '18px',
    borderBottom: `1px solid ${theme.palette.divider}`
  },

  ampClosed: {
    display: 'none'
  }
});

@withStyles(styles, { name: 'RSFDrawer' })
@inject(({ app, ampStateId }) => ({ amp: app && app.amp, ampStateId }))
export default class Drawer extends Component {

  static propTypes = {
    /**
     * Set to false to hide the close button. Defaults to true
     */
    showCloseButton: PropTypes.bool,

    /**
     * Called when the user closes the drawer
     */
    onRequestClose: PropTypes.func.isRequired,

    /**
     * The title to display at the top of the drawer
     */
    title: PropTypes.oneOfType([ PropTypes.string, PropTypes.element ]),

    /**
     * Set to true to automatically add padding to the body when the drawer
     * is open so that the user is able to scroll and see all of the body content.
     * Defaults to false.
     */
    autoAdjustBodyPadding: PropTypes.bool,

    /**
     * Props to apply to the closeButton
     */
    closeButtonProps: PropTypes.object,

    /**
     * The name of a property the amp-state to bind to the closed state of the drawer.
     */
    ampBindClosed: PropTypes.string
  }

  static defaultProps = {
    showCloseButton: true,
    autoAdjustBodyPadding: false,
    closeButtonProps: {}
  }

  constructor() {
    super();
    this.drawer = React.createRef()
  }

  render() {
    const { amp, ampStateId, ampBindClosed, variant, closeButtonProps, showCloseButton, open, onRequestClose, title, children, classes, autoAdjustBodyPadding, ...rest } = this.props

    return (
      <MUIDrawer 
        anchor="bottom" 
        classes={{
          paper: classnames({
            [classes.paper]: true,
            [classes.ampClosed]: amp && !open
          })
        }}
        amp-bind={ampBindClosed ? `class=>${ampStateId}.${ampBindClosed} ? '${classes.ampClosed}' : null` : null} 
        open={(amp && variant === "temporary") || open}
        variant={variant}
        {...rest}
      >
        <div ref={this.drawer}>
          { title && (
            <div className={classes.title}>
              <Typography variant="h6" component="div">{title}</Typography>
            </div>
          )}

          { showCloseButton && (
            <Button 
              variant="fab" 
              color="primary" 
              className={classes.closeButton} 
              onClick={this.closeDrawer} 
              style={{ display: open ? '' : 'none' }}
              on={ampBindClosed ? `tap:AMP.setState({ ${ampStateId}: { ${ampBindClosed}: true }})` : null}
              {...closeButtonProps}
            >
              <Close/>
            </Button>
          )}

          { children }
        </div>
      </MUIDrawer>
    )
  }

  // add body padding equal to drawer height
  componentDidMount() {
    this.setPadding()
    const el = this.drawer.current

    if (this.props.autoAdjustBodyPadding && el) {
      this.drawerResize = new ResizeObserver(() => {
        document.body.style.paddingBottom = el && el.clientHeight + 'px'
      })

      this.drawerResize.observe(el)
    }
  }

  componentWillUnmount() {
    const el = this.drawer.current

    if (this.drawerResize && el) {
      this.drawerResize.unobserve(el)
    }
  }

  // if value of open property changes, recalcuate drawer height/body padding
  componentWillReceiveProps(nextProps) {
    if (nextProps.open && !this.props.open) {
        this.setPadding()
    } else if (this.props.open && !nextProps.open) {
        this.closeDrawer()
    }
  }

  setPadding() {
    if (this.props.autoAdjustBodyPadding) {
      requestAnimationFrame(() => {
        const el = this.drawer.current
        document.body.style.paddingBottom = el && el.clientHeight + 'px'
      })
    }
  }

  closeDrawer = () => {
    const { onRequestClose } = this.props
    onRequestClose()
    document.body.style.paddingBottom = 0
  }

}
