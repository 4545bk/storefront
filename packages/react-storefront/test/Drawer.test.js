/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount, shallow } from 'enzyme'
import Drawer from '../src/Drawer'
import AmpState from '../src/amp/AmpState'
import TestProvider from './TestProvider'

describe('Drawer', () => {

  it('accepts a className prop', () => {
    expect(mount(
      <TestProvider>
        <Drawer className="myClassName" onRequestClose={Function.prototype}/>
      </TestProvider>
    )).toMatchSnapshot()
  })

  it('accepts a paper class', () => {
    expect(mount(
      <TestProvider>
        <Drawer classes={{ paper: 'myPaperClass' }} onRequestClose={Function.prototype}/>
      </TestProvider>)).toMatchSnapshot()
  })

  it('uses a div for title', () => {
    expect(mount(
      <TestProvider>
        <Drawer open title="Title" onRequestClose={Function.prototype}/>
      </TestProvider>
    )).toMatchSnapshot()
  })

  it('spreads props to the underlying MuiDrawer', () => {
    // We MUST use Enzyme here because the DOM is used
    // And mount causes a invalid string length error. This may
    // be related to the enzyme-to-json library and may
    // be fixed in the future. For now, shallow will do.
    expect(mount(
      <TestProvider>
        <Drawer open onRequestClose={Function.prototype}/>
      </TestProvider>
    )).toMatchSnapshot()
  })

  it('sets up amp-bind when ampBindClosed is set', () => {
    expect(mount(
      <TestProvider app={{ amp: true }}>
        <AmpState id="myState">
          <Drawer ampBindClosed="closed" onRequestClose={Function.prototype}/>
        </AmpState>
      </TestProvider>
    )).toMatchSnapshot()
  })
})