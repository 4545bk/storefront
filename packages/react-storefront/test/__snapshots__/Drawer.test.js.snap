// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer accepts a className prop 1`] = `
<TestProvider>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-Drawer) className="myClassName" onRequestClose={[Function]}>
        <inject-Drawer className="myClassName" onRequestClose={[Function]} classes={{...}}>
          <Drawer className="myClassName" onRequestClose={[Function]} classes={{...}} amp={false} ampStateId={[undefined]} showCloseButton={true} autoAdjustBodyPadding={false} closeButtonProps={{...}}>
            <WithStyles(Drawer) anchor="bottom" classes={{...}} amp-bind={{...}} open={[undefined]} variant={[undefined]} className="myClassName">
              <Drawer anchor="bottom" amp-bind={{...}} open={false} variant="temporary" className="myClassName" theme={{...}} classes={{...}} elevation={16} transitionDuration={{...}}>
                <WithStyles(Modal) BackdropProps={{...}} className="MuiDrawer-root-7 MuiDrawer-modal-18 myClassName" open={false} onClose={[undefined]} amp-bind={{...}}>
                  <Modal BackdropProps={{...}} className="MuiDrawer-root-7 MuiDrawer-modal-18 myClassName" open={false} onClose={[undefined]} amp-bind={{...}} classes={{...}} BackdropComponent={[Function: WithStyles]} disableAutoFocus={false} disableBackdropClick={false} disableEnforceFocus={false} disableEscapeKeyDown={false} disablePortal={false} disableRestoreFocus={false} hideBackdrop={false} keepMounted={false} manager={{...}} />
                </WithStyles(Modal)>
              </Drawer>
            </WithStyles(Drawer)>
          </Drawer>
        </inject-Drawer>
      </WithStyles(inject-Drawer)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`Drawer accepts a paper class 1`] = `
<TestProvider>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-Drawer) classes={{...}} onRequestClose={[Function]}>
        <inject-Drawer onRequestClose={[Function]} classes={{...}}>
          <Drawer onRequestClose={[Function]} classes={{...}} amp={false} ampStateId={[undefined]} showCloseButton={true} autoAdjustBodyPadding={false} closeButtonProps={{...}}>
            <WithStyles(Drawer) anchor="bottom" classes={{...}} amp-bind={{...}} open={[undefined]} variant={[undefined]}>
              <Drawer anchor="bottom" amp-bind={{...}} open={false} variant="temporary" theme={{...}} classes={{...}} elevation={16} transitionDuration={{...}}>
                <WithStyles(Modal) BackdropProps={{...}} className="MuiDrawer-root-27 MuiDrawer-modal-38" open={false} onClose={[undefined]} amp-bind={{...}}>
                  <Modal BackdropProps={{...}} className="MuiDrawer-root-27 MuiDrawer-modal-38" open={false} onClose={[undefined]} amp-bind={{...}} classes={{...}} BackdropComponent={[Function: WithStyles]} disableAutoFocus={false} disableBackdropClick={false} disableEnforceFocus={false} disableEscapeKeyDown={false} disablePortal={false} disableRestoreFocus={false} hideBackdrop={false} keepMounted={false} manager={{...}} />
                </WithStyles(Modal)>
              </Drawer>
            </WithStyles(Drawer)>
          </Drawer>
        </inject-Drawer>
      </WithStyles(inject-Drawer)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`Drawer sets up amp-bind when ampBindClosed is set 1`] = `
<TestProvider app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <inject-AmpState id="myState">
        <AmpState id="myState" amp={true} initialState={{...}}>
          <Provider ampStateId="myState">
            <HelmetWrapper defer={true} encodeSpecialCharacters={true}>
              <SideEffect(NullComponent) defer={true} encodeSpecialCharacters={true} script={{...}}>
                <NullComponent defer={true} encodeSpecialCharacters={true} script={{...}} />
              </SideEffect(NullComponent)>
            </HelmetWrapper>
            <amp-state id="myState">
              <script type="application/json" dangerouslySetInnerHTML={{...}} />
            </amp-state>
            <WithStyles(inject-Drawer) ampBindClosed="closed" onRequestClose={[Function]}>
              <inject-Drawer ampBindClosed="closed" onRequestClose={[Function]} classes={{...}}>
                <Drawer ampBindClosed="closed" onRequestClose={[Function]} classes={{...}} amp={true} ampStateId="myState" showCloseButton={true} autoAdjustBodyPadding={false} closeButtonProps={{...}}>
                  <WithStyles(Drawer) anchor="bottom" classes={{...}} amp-bind="class=>myState.closed ? \\'RSFDrawer-ampClosed-269\\' : null" open={[undefined]} variant={[undefined]}>
                    <Drawer anchor="bottom" amp-bind="class=>myState.closed ? \\'RSFDrawer-ampClosed-269\\' : null" open={false} variant="temporary" theme={{...}} classes={{...}} elevation={16} transitionDuration={{...}}>
                      <WithStyles(Modal) BackdropProps={{...}} className="MuiDrawer-root-270 MuiDrawer-modal-281" open={false} onClose={[undefined]} amp-bind="class=>myState.closed ? \\'RSFDrawer-ampClosed-269\\' : null">
                        <Modal BackdropProps={{...}} className="MuiDrawer-root-270 MuiDrawer-modal-281" open={false} onClose={[undefined]} amp-bind="class=>myState.closed ? \\'RSFDrawer-ampClosed-269\\' : null" classes={{...}} BackdropComponent={[Function: WithStyles]} disableAutoFocus={false} disableBackdropClick={false} disableEnforceFocus={false} disableEscapeKeyDown={false} disablePortal={false} disableRestoreFocus={false} hideBackdrop={false} keepMounted={false} manager={{...}} />
                      </WithStyles(Modal)>
                    </Drawer>
                  </WithStyles(Drawer)>
                </Drawer>
              </inject-Drawer>
            </WithStyles(inject-Drawer)>
          </Provider>
        </AmpState>
      </inject-AmpState>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`Drawer spreads props to the underlying MuiDrawer 1`] = `
<TestProvider>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-Drawer) open={true} onRequestClose={[Function]}>
        <inject-Drawer open={true} onRequestClose={[Function]} classes={{...}}>
          <Drawer open={true} onRequestClose={[Function]} classes={{...}} amp={false} ampStateId={[undefined]} showCloseButton={true} autoAdjustBodyPadding={false} closeButtonProps={{...}}>
            <WithStyles(Drawer) anchor="bottom" classes={{...}} amp-bind={{...}} open={true} variant={[undefined]}>
              <Drawer anchor="bottom" amp-bind={{...}} open={true} variant="temporary" theme={{...}} classes={{...}} elevation={16} transitionDuration={{...}}>
                <WithStyles(Modal) BackdropProps={{...}} className="MuiDrawer-root-176 MuiDrawer-modal-187" open={true} onClose={[undefined]} amp-bind={{...}}>
                  <Modal BackdropProps={{...}} className="MuiDrawer-root-176 MuiDrawer-modal-187" open={true} onClose={[undefined]} amp-bind={{...}} classes={{...}} BackdropComponent={[Function: WithStyles]} disableAutoFocus={false} disableBackdropClick={false} disableEnforceFocus={false} disableEscapeKeyDown={false} disablePortal={false} disableRestoreFocus={false} hideBackdrop={false} keepMounted={false} manager={{...}}>
                    <Portal container={[undefined]} disablePortal={false} onRendered={[Function]}>
                      <Portal containerInfo={{...}}>
                        <div className="MuiModal-root-188 MuiDrawer-root-176 MuiDrawer-modal-187" amp-bind={{...}}>
                          <WithStyles(Backdrop) open={true} onClick={[Function]} transitionDuration={{...}}>
                            <Backdrop open={true} onClick={[Function]} transitionDuration={{...}} classes={{...}} invisible={false}>
                              <WithTheme(Fade) in={true} timeout={{...}} onClick={[Function]}>
                                <Fade theme={{...}} in={true} timeout={{...}} onClick={[Function]}>
                                  <Transition appear={true} onEnter={[Function]} onExit={[Function]} in={true} timeout={{...}} onClick={[Function]} mountOnEnter={false} unmountOnExit={false} enter={true} exit={true} onEntering={[Function: noop]} onEntered={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]}>
                                    <div className="MuiBackdrop-root-190" aria-hidden="true" style={{...}} onClick={[Function]} />
                                  </Transition>
                                </Fade>
                              </WithTheme(Fade)>
                            </Backdrop>
                          </WithStyles(Backdrop)>
                          <RootRef rootRef={[Function]}>
                            <WithTheme(Slide) in={true} direction="up" timeout={{...}} appear={false} onExited={[Function: chainedFunction]} role="document" tabIndex="-1">
                              <Slide theme={{...}} in={true} direction="up" timeout={{...}} appear={false} onExited={[Function: chainedFunction]} role="document" tabIndex="-1">
                                <EventListener target="window" onResize={[Function: debounced]}>
                                  <Transition onEnter={[Function]} onEntering={[Function]} onExit={[Function]} onExited={[Function]} appear={false} style={{...}} in={true} direction="up" timeout={{...}} role="document" tabIndex="-1" mountOnEnter={false} unmountOnExit={false} enter={true} exit={true} onEntered={[Function: noop]} onExiting={[Function: noop]}>
                                    <WithStyles(Paper) elevation={16} square={true} className="MuiDrawer-paper-178 RSFDrawer-paper-173 MuiDrawer-paperAnchorBottom-182" style={{...}} direction="up" role="document" tabIndex="-1">
                                      <Paper elevation={16} square={true} className="MuiDrawer-paper-178 RSFDrawer-paper-173 MuiDrawer-paperAnchorBottom-182" style={{...}} direction="up" role="document" tabIndex="-1" classes={{...}} component="div">
                                        <div className="MuiPaper-root-192 MuiPaper-elevation16-210 MuiDrawer-paper-178 RSFDrawer-paper-173 MuiDrawer-paperAnchorBottom-182" style={{...}} direction="up" role="document" tabIndex="-1">
                                          <div className="RSFDrawer-container-171">
                                            <WithStyles(Button) variant="fab" color="primary" className="RSFDrawer-closeButton-170" onClick={[Function]} style={{...}} on={{...}}>
                                              <Button variant="fab" color="primary" className="RSFDrawer-closeButton-170" onClick={[Function]} style={{...}} on={{...}} classes={{...}} component="button" disabled={false} disableFocusRipple={false} fullWidth={false} mini={false} size="medium" type="button">
                                                <WithStyles(ButtonBase) className="MuiButton-root-219 MuiButton-fab-236 MuiButton-contained-230 MuiButton-containedPrimary-231 MuiButton-raised-233 MuiButton-raisedPrimary-234 RSFDrawer-closeButton-170" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-238" onClick={[Function]} style={{...}} on={{...}} component="button" type="button">
                                                  <ButtonBase className="MuiButton-root-219 MuiButton-fab-236 MuiButton-contained-230 MuiButton-containedPrimary-231 MuiButton-raised-233 MuiButton-raisedPrimary-234 RSFDrawer-closeButton-170" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-238" onClick={[Function]} style={{...}} on={{...}} component="button" type="button" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} tabIndex="0">
                                                    <button className="MuiButtonBase-root-245 MuiButton-root-219 MuiButton-fab-236 MuiButton-contained-230 MuiButton-containedPrimary-231 MuiButton-raised-233 MuiButton-raisedPrimary-234 RSFDrawer-closeButton-170" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" type="button" disabled={false} onClick={[Function]} style={{...}} on={{...}}>
                                                      <span className="MuiButton-label-220">
                                                        <pure(CloseIcon)>
                                                          <CloseIcon>
                                                            <WithStyles(SvgIcon)>
                                                              <SvgIcon classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                                <svg className="MuiSvgIcon-root-248" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                                  <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
                                                                  <path fill="none" d="M0 0h24v24H0z" />
                                                                </svg>
                                                              </SvgIcon>
                                                            </WithStyles(SvgIcon)>
                                                          </CloseIcon>
                                                        </pure(CloseIcon)>
                                                      </span>
                                                      <NoSsr defer={false} fallback={{...}}>
                                                        <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                                                          <TouchRipple center={false} classes={{...}}>
                                                            <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-257" childFactory={[Function: childFactory]}>
                                                              <span className="MuiTouchRipple-root-257" />
                                                            </TransitionGroup>
                                                          </TouchRipple>
                                                        </WithStyles(TouchRipple)>
                                                      </NoSsr>
                                                    </button>
                                                  </ButtonBase>
                                                </WithStyles(ButtonBase)>
                                              </Button>
                                            </WithStyles(Button)>
                                            <div className="RSFDrawer-content-172" />
                                          </div>
                                        </div>
                                      </Paper>
                                    </WithStyles(Paper)>
                                  </Transition>
                                </EventListener>
                              </Slide>
                            </WithTheme(Slide)>
                          </RootRef>
                        </div>
                      </Portal>
                    </Portal>
                  </Modal>
                </WithStyles(Modal)>
              </Drawer>
            </WithStyles(Drawer)>
          </Drawer>
        </inject-Drawer>
      </WithStyles(inject-Drawer)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`Drawer uses a div for title 1`] = `
<TestProvider>
  <Provider router={{...}} app={{...}} history={{...}}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-Drawer) open={true} title="Title" onRequestClose={[Function]}>
        <inject-Drawer open={true} title="Title" onRequestClose={[Function]} classes={{...}}>
          <Drawer open={true} title="Title" onRequestClose={[Function]} classes={{...}} amp={false} ampStateId={[undefined]} showCloseButton={true} autoAdjustBodyPadding={false} closeButtonProps={{...}}>
            <WithStyles(Drawer) anchor="bottom" classes={{...}} amp-bind={{...}} open={true} variant={[undefined]}>
              <Drawer anchor="bottom" amp-bind={{...}} open={true} variant="temporary" theme={{...}} classes={{...}} elevation={16} transitionDuration={{...}}>
                <WithStyles(Modal) BackdropProps={{...}} className="MuiDrawer-root-47 MuiDrawer-modal-58" open={true} onClose={[undefined]} amp-bind={{...}}>
                  <Modal BackdropProps={{...}} className="MuiDrawer-root-47 MuiDrawer-modal-58" open={true} onClose={[undefined]} amp-bind={{...}} classes={{...}} BackdropComponent={[Function: WithStyles]} disableAutoFocus={false} disableBackdropClick={false} disableEnforceFocus={false} disableEscapeKeyDown={false} disablePortal={false} disableRestoreFocus={false} hideBackdrop={false} keepMounted={false} manager={{...}}>
                    <Portal container={[undefined]} disablePortal={false} onRendered={[Function]}>
                      <Portal containerInfo={{...}}>
                        <div className="MuiModal-root-59 MuiDrawer-root-47 MuiDrawer-modal-58" amp-bind={{...}}>
                          <WithStyles(Backdrop) open={true} onClick={[Function]} transitionDuration={{...}}>
                            <Backdrop open={true} onClick={[Function]} transitionDuration={{...}} classes={{...}} invisible={false}>
                              <WithTheme(Fade) in={true} timeout={{...}} onClick={[Function]}>
                                <Fade theme={{...}} in={true} timeout={{...}} onClick={[Function]}>
                                  <Transition appear={true} onEnter={[Function]} onExit={[Function]} in={true} timeout={{...}} onClick={[Function]} mountOnEnter={false} unmountOnExit={false} enter={true} exit={true} onEntering={[Function: noop]} onEntered={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]}>
                                    <div className="MuiBackdrop-root-61" aria-hidden="true" style={{...}} onClick={[Function]} />
                                  </Transition>
                                </Fade>
                              </WithTheme(Fade)>
                            </Backdrop>
                          </WithStyles(Backdrop)>
                          <RootRef rootRef={[Function]}>
                            <WithTheme(Slide) in={true} direction="up" timeout={{...}} appear={false} onExited={[Function: chainedFunction]} role="document" tabIndex="-1">
                              <Slide theme={{...}} in={true} direction="up" timeout={{...}} appear={false} onExited={[Function: chainedFunction]} role="document" tabIndex="-1">
                                <EventListener target="window" onResize={[Function: debounced]}>
                                  <Transition onEnter={[Function]} onEntering={[Function]} onExit={[Function]} onExited={[Function]} appear={false} style={{...}} in={true} direction="up" timeout={{...}} role="document" tabIndex="-1" mountOnEnter={false} unmountOnExit={false} enter={true} exit={true} onEntered={[Function: noop]} onExiting={[Function: noop]}>
                                    <WithStyles(Paper) elevation={16} square={true} className="MuiDrawer-paper-49 RSFDrawer-paper-44 MuiDrawer-paperAnchorBottom-53" style={{...}} direction="up" role="document" tabIndex="-1">
                                      <Paper elevation={16} square={true} className="MuiDrawer-paper-49 RSFDrawer-paper-44 MuiDrawer-paperAnchorBottom-53" style={{...}} direction="up" role="document" tabIndex="-1" classes={{...}} component="div">
                                        <div className="MuiPaper-root-63 MuiPaper-elevation16-81 MuiDrawer-paper-49 RSFDrawer-paper-44 MuiDrawer-paperAnchorBottom-53" style={{...}} direction="up" role="document" tabIndex="-1">
                                          <div className="RSFDrawer-container-42">
                                            <WithStyles(Typography) className="RSFDrawer-title-45" variant="h6" component="div">
                                              <Typography className="RSFDrawer-title-45" variant="h6" component="div" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                                                <div className="MuiTypography-root-90 MuiTypography-h6-107 RSFDrawer-title-45">
                                                  Title
                                                </div>
                                              </Typography>
                                            </WithStyles(Typography)>
                                            <WithStyles(Button) variant="fab" color="primary" className="RSFDrawer-closeButton-41" onClick={[Function]} style={{...}} on={{...}}>
                                              <Button variant="fab" color="primary" className="RSFDrawer-closeButton-41" onClick={[Function]} style={{...}} on={{...}} classes={{...}} component="button" disabled={false} disableFocusRipple={false} fullWidth={false} mini={false} size="medium" type="button">
                                                <WithStyles(ButtonBase) className="MuiButton-root-125 MuiButton-fab-142 MuiButton-contained-136 MuiButton-containedPrimary-137 MuiButton-raised-139 MuiButton-raisedPrimary-140 RSFDrawer-closeButton-41" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-144" onClick={[Function]} style={{...}} on={{...}} component="button" type="button">
                                                  <ButtonBase className="MuiButton-root-125 MuiButton-fab-142 MuiButton-contained-136 MuiButton-containedPrimary-137 MuiButton-raised-139 MuiButton-raisedPrimary-140 RSFDrawer-closeButton-41" disabled={false} focusRipple={true} focusVisibleClassName="MuiButton-focusVisible-144" onClick={[Function]} style={{...}} on={{...}} component="button" type="button" classes={{...}} centerRipple={false} disableRipple={false} disableTouchRipple={false} tabIndex="0">
                                                    <button className="MuiButtonBase-root-151 MuiButton-root-125 MuiButton-fab-142 MuiButton-contained-136 MuiButton-containedPrimary-137 MuiButton-raised-139 MuiButton-raisedPrimary-140 RSFDrawer-closeButton-41" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex="0" type="button" disabled={false} onClick={[Function]} style={{...}} on={{...}}>
                                                      <span className="MuiButton-label-126">
                                                        <pure(CloseIcon)>
                                                          <CloseIcon>
                                                            <WithStyles(SvgIcon)>
                                                              <SvgIcon classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                                                <svg className="MuiSvgIcon-root-154" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                                                  <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
                                                                  <path fill="none" d="M0 0h24v24H0z" />
                                                                </svg>
                                                              </SvgIcon>
                                                            </WithStyles(SvgIcon)>
                                                          </CloseIcon>
                                                        </pure(CloseIcon)>
                                                      </span>
                                                      <NoSsr defer={false} fallback={{...}}>
                                                        <WithStyles(TouchRipple) innerRef={[Function]} center={false}>
                                                          <TouchRipple center={false} classes={{...}}>
                                                            <TransitionGroup component="span" enter={true} exit={true} className="MuiTouchRipple-root-163" childFactory={[Function: childFactory]}>
                                                              <span className="MuiTouchRipple-root-163" />
                                                            </TransitionGroup>
                                                          </TouchRipple>
                                                        </WithStyles(TouchRipple)>
                                                      </NoSsr>
                                                    </button>
                                                  </ButtonBase>
                                                </WithStyles(ButtonBase)>
                                              </Button>
                                            </WithStyles(Button)>
                                            <div className="RSFDrawer-content-43" />
                                          </div>
                                        </div>
                                      </Paper>
                                    </WithStyles(Paper)>
                                  </Transition>
                                </EventListener>
                              </Slide>
                            </WithTheme(Slide)>
                          </RootRef>
                        </div>
                      </Portal>
                    </Portal>
                  </Modal>
                </WithStyles(Modal)>
              </Drawer>
            </WithStyles(Drawer)>
          </Drawer>
        </inject-Drawer>
      </WithStyles(inject-Drawer)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
